Dockerfile

```diff
@@ -1,15 +1,37 @@
-FROM python:3.9
+FROM python:3.9-alpine as base
+
+ENV PYTHONFAULTHANDLER=1 \
+    PYTHONHASHSEED=random \
+    PYTHONUNBUFFERED=1
+
+ENV VENV_PATH="/venv"
+
+ENV PATH="$VENV_PATH/bin:$PATH"
 
 WORKDIR /app
 
-RUN apt-get update && apt-get upgrade -y
+FROM base as builder
 
-RUN pip install --upgrade pip wheel setuptools poetry
+ENV PIP_DEFAULT_TIMEOUT=100 \
+    PIP_DISABLE_PIP_VERSION_CHECK=1 \
+    PIP_NO_CACHE_DIR=1 \
+    POETRY_VERSION=1.0.5
 
-ENV POETRY_VIRTUALENVS_IN_PROJECT true
+ENV CRYPTOGRAPHY_DONT_BUILD_RUST=1
+RUN apk add --no-cache gcc libffi-dev musl-dev postgresql-dev postgresql-dev musl-dev zlib zlib-dev jpeg-dev
```

Comment:
The thought of using alpine came to my mind earlier, but `tgcf` did not successfully run on alpine. Due to these missing dependencies. 

Manually maintaining these deps in Dockerfile is a hard job for me. Some missing deps can affect some parts of the code. I am afraid that something will break.

In the future, I will be also adding the watermark and OCR plugins, in the default package. So in the Dockerfile, I need to add FFmpeg and tesseract. Installing them in alpine may be a hard job.

The size of the image does not affect run time performance, so I am less interested in optimizing it. 

Most of my users, run tgcf on their computer, installed via pip. And many deploy to Heroku. 

Using alpine may introduce hard to debug problems in the future. I don't have a 100% test coverage, that will allow me to reliably test this image.

So, my suggestion, is that rename this Dockerfile, to `Dockerfile.small`. I will publish this as a separate tag, `tgcf:small`. So this change will be "opt-in" for the end users.

After a sufficient amount of testing (3 to 4 months, by multiple users), we may make this Dockerfile the default.

Reply:
oh yes, I initially developed smaller one for my local repo. I will push that way.

> Manually maintaining these deps in Dockerfile is a hard job for me. Some missing deps can affect some parts of the code. I am afraid that something will break.
true but little effort but onetime.

 also pushed https://hub.docker.com/repository/docker/nbmaiti/tgcf for multi platforms.

Reply:
Pushed new patch.

Please add the build for minimal in your wiki relevant place

docker build . -f Dockerfile.minimal -t imagename